### ISSUE TITLE
Add binding and tests for IMessageReceiver in ServiceBusTrigger #2218

### ISSUE THREAD

**Issue by StannieV** [Azure/azure-webjobs-sdk#2365]
As mentioned in [issue 2222](https://github.com/Azure/azure-webjobs-sdk/issues/2222), I'm interested in 'Support interface IMessageReceiver via Dependency Injection'. Please add this feature so unit testing is easier.

**Comment by kev24uk** [comment]
I would also like to see this implemented!

**Comment by eminencegrs** [comment]
I absolutely agree with @StannieV. It allows me to simplify unit testing.

**Comment by bradshaw84nz** [comment]
I would also like to see this implemented

**Comment by luisega** [comment]
+1 to this feature. It will be great for unit testing

**Comment by nickwilsonr** [comment]
+1 to this feature as well

**Comment by JBMCeridian** [comment]
+1, very hard to unit test in its current implementation

**Comment by evilpilaf** [comment]
I agree, this should be the default, either with IMessageReceiver or another interface, but the triggers shouldn't depend on concrete types.

**Comment by gornikp** [comment]
+1, Please add it! It is like basic functionality!

**Comment by kdr213** [comment]
Unit testing in this scenario is super painful today. This would be a major improvement.

**Comment by siparsons** [comment]
+1 seems like a major oversight that this is not there.

**Comment by dhtek** [comment]
+1 We also need to unit test and it's very hard today !

**Comment by jakub-klapka** [comment]
+1 Unit testing function main method (which is the case for all functions out there) is painfully verbose without this.

**Comment by ttjackott** [comment]
Please add this! Desperate for this for the sake of our unit tests!

**Comment by AlexEngblom** [comment]
+1 unit testing without this is painful.

**Comment by AndrewMorris-2** [comment]
This feature would be extremely useful for unit testing.

**Comment by lansal15** [comment]
+1

**Comment by markhamannpolaris** [comment]
+1

**Comment by tgursk** [comment]
+1

**Comment by brandonBotz** [comment]
+1

**Comment by jfren484** [comment]
+1

**Comment by v-anvari** [comment]
@alrod , Is this a feature we can add?

**Comment by jjcv2304** [comment]
There is a solution proposal on stackoverflow https://stackoverflow.com/questions/64904741/how-to-unit-test-when-microsoft-azure-servicebus-core-messagereceiver-is-involve#comment114768525_64904741
But I'm not sure of possible side effects when casting down to the interface, Does someone knows if that solution has any inconvenient, aside from having to add more code?

**Comment by StannieV** [comment]
Perhaps [my suggestion](https://github.com/Azure/azure-webjobs-sdk/pull/2218#issuecomment-647584346) in #2218 is also an option.

**Comment by jjcv2304** [comment]
@StannieV I like your solution better than the one on stackoverflow. Would you like to add it to stackoverflow? I would accept it as the right answer

**Comment by StannieV** [comment]
Done
